.github/workflows/reusable-go-fmt-vet.yml

```diff
@@ -0,0 +1,52 @@
+name: go-fmt-and-vet 
+
+on:
+  workflow_call:
+    inputs:
+      directory:
+        required: true
+        type: string
+      go-version:
+        required: true
+        type: string
+
+jobs:
+  a:
```

Comment:
I'm curious if we can name it a bit more descriptively? maybe `go-fmt-and-vet`?

Reply:
This one is funky and I couldn't think of a nice way to do it. If you take a look at the visual diagram of the workflows (https://github.com/hashicorp/consul-k8s/actions/runs/1953797104) you will see that several of the job names are:

`go-fmt-vet-cli / a` or `go-fmt-vet-helm-gen / a`. This is because it is concatenating `<workflow job name> / <reusable job name>`. The constraints are that <workflow job name> has to be unique and I could not do something like `cli / go-fmt-vet` as the next `cli` job name in the workflow would be in conflict. Also, the reusable workflow has to have a job name or it will not work...

Reply:
Yeah, naming is hard 😄 

What if we name it `go-fmt-and-vet` or `go-fmt-vet` or `fmt-vet`? Then it will show as:
- `go-fmt-vet-cli/go-fmt-and-vet`
- `go-fmt-vet-cli/go-fmt-vet`
- `go-fmt-vet-cli/fmt-vet` 

Another idea: what if we name them something also generic like `job` but slightly more specific than `a`? Then it'll be `go-fmt-vet-cli/job`.

I think it'd be nice to have more specific/descriptive names if at all possible. WDYT?

Reply:
Naming is terribly hard! 

I think 'job' may be a key word and will cause GHA to fail.  How about `lint` since I moved go-fmt-vet to use golangci-lint?